ddl/partition.go

```diff
@@ -1470,7 +1470,13 @@ func onAlterTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
 	}
 
 	for i, rule := range rules {
-		rule.ID = fmt.Sprintf("%d_%d_%d_%d_%d", job.SchemaID, tblInfo.ID, partitionID, job.ID, i)
+		if rule.Action == placement.RuleOpDel {
+			rule.ID = fmt.Sprintf("%d_%d_%s", tblInfo.ID, partitionID, rule.Role)
+		} else if len(rule.LabelConstraints) == 0 {
+			rule.ID = fmt.Sprintf("%d_%d_%s", tblInfo.ID, partitionID, rule.Role)
+		} else {
+			rule.ID = fmt.Sprintf("%d_%d_%s_%d_%d", tblInfo.ID, partitionID, rule.Role, job.ID, i)
```

Comment:
Why add table ID?

Reply:
It just works. But as you mentioned. I add the schema ID back. The full path is to improve efficiency of PD. One deletion by prefix means a full iteration of all rules. A common prefix can archive this action by only one deletion.
It is of course possible to iterate tableID from database, and create several deletions. But it is not the fastest way.

Reply:
`t` and `p` is to distinguish sub rules from its own rules.